### ISSUE TITLE
Staging and Production:  User Access Token Expired While Active in App

### ISSUE THREAD

**Issue by marcynadeau** [department-of-veterans-affairs/va.gov-team#50013]
This bug is impacting testing in staging because it happens all too frequently.  It is so FRUSTRATING to be in the middle of testing and get this token thing because I have to start all over again!  

I have seen this issue in production.

At some point in the apps while the user is active and progress through the app the token expires.  

Steps to reproduce
1.	Log into VAOS.  Continue through direct scheduling or request flow.
2.	Pause on one of the pages (location or clinic) or go back and forth through different options/screens. The key here is to not submit or schedule anything.   Not sure of the time interval here.
3.	At some point, complete the process you started, complete the flow.  User is able to navigate within VAOS.  No errors displays as user navigates through VAOS screens.  
4.	Submit request or schedule appointment
5.	Expected response:  Appointment/ request is submitted.  
6.	Actual response:  Veteran gets an error.  In dev tools show access token expired. 
 
What has happened is the user has been logged out in the background but in the UI the user shows as still logged in and can navigate through the application. 

![image.png](https://images.zenhubusercontent.com/5f2d82e4a65f7435058cfd75/35bd94a4-e87f-46b7-9440-f57da94eb5a9)

**Comment by ldelacosta** [comment]
@cferris32 @simiadebowale - When you have a chance, can you look at this ticket?

**Comment by simiadebowale** [comment]
updating this ticket with platform support ticket thread: https://dsva.slack.com/archives/CBU0KDSB1/p1669751971819239

**Comment by simiadebowale** [comment]
[PR to fix this](https://github.com/department-of-veterans-affairs/vets-website/pull/23011)

**Comment by simiadebowale** [comment]
@ldelacosta this is ready for validation.

**Comment by ldelacosta** [comment]
@marcynadeau - This ticket is ready to be validated in staging. Let us know if you have any questions. Thanks!

**Comment by ldelacosta** [comment]
Adding @ryanshaw in case SQA has any issues.

**Comment by ymunipalli** [comment]
![image](https://user-images.githubusercontent.com/76223983/218177156-9befcb7a-dfa9-4bc0-9c0d-248d19da53e7.png)
![image](https://user-images.githubusercontent.com/76223983/218177290-3317ca1c-867f-4c41-8720-2ad1da34285c.png)
![image](https://user-images.githubusercontent.com/76223983/218177425-baba3dd8-ddb1-4a03-bc50-db7b70d3a2e2.png)

**Comment by ryanshaw** [comment]
@ymunipalli @ldelacosta I am able to reproduce this. Will do some digging.

**Comment by ryanshaw** [comment]
@ymunipalli @ldelacosta There's an update on [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/23011#issuecomment-1426284116)

**Comment by ldelacosta** [comment]
Testing is on hold until the new automatic refresh PR is available in staging.

**Comment by ryanshaw** [comment]
@ldelacosta Looks like the [PR](https://github.com/department-of-veterans-affairs/vets-website/pull/23263) has been merged in, I'll take a look at staging.

**Comment by ryanshaw** [comment]
I've verified that the above referenced PR does fix this on staging.

**Comment by ymunipalli** [comment]
@ryanshaw @marcynadeau @ldelacosta I am still getting the 403 errors. Let me provide the details here: 
1. Login as grethel.g.petryniec@id.me
2. Go into the VAOS scheduling website.
3. Click on Start Scheduling.
4. Enter the information for a request for a CC flow.
5. Enter English as your language preference.
6. Stop when you are about to request appointment and go away for a few minutes- like 8 minutes.
7. Click on Request appointment. 
8. Go to devtools and notice a 403 error message.
Expected behavior is that no 403 messages would display.
Actual behavior is that there is a 403 error in devtools.

![image](https://user-images.githubusercontent.com/76223983/219381244-df5fed2c-fbd5-4315-9517-b0aea36d3c94.png)
![image](https://user-images.githubusercontent.com/76223983/219381427-ab770448-8c97-4a8c-8142-8c3093a0128d.png)
![image](https://user-images.githubusercontent.com/76223983/219381593-3f3bb84a-72c1-4a48-9a0f-628daceee063.png)

**Comment by ryanshaw** [comment]
This issue has been resolved with this [PR](https://github.com/department-of-veterans-affairs/vets-website/pull/23263).

However, it should be noted that one may still see `403` errors. The fix for the timeout issue keys off the `403` token expiration error and initiates a call to the `/refresh` endpoint to refresh the token and then proceeds.

I've attached two screenshots. The first one shows a user stopped at the confirmation page in an appointment flow. The second screenshot shows what happens after a user allows their token to expire (after approximately six minutes) but continues the flow by confirming the appointment. A `403` error is returned, `/refresh` endpoint is called and then the appointment is confirmed.

We may need to adjust how we are looking at these specific `403` errors within our monitoring and/or analytics setups.

<img width="1751" alt="Screenshot 2023-02-17 at 9 14 01 AM" src="https://user-images.githubusercontent.com/587812/219724259-da8b564f-cef8-43d5-a6db-cadb51b994b7.png">

<img width="1745" alt="Screenshot 2023-02-17 at 9 21 32 AM" src="https://user-images.githubusercontent.com/587812/219724417-d4be1042-c69c-4ef5-a9c8-7c6f57bf2575.png">
